File: libpandadna/DNASuitPath.py
class DNASuitPath:
    def __init__(self):
        self.suitPoints = []

    def getNumPoints(self):
        return len(self.suitPoints)

    def getPointIndex(self, pointIndex):
        return self.suitPoints[pointIndex].getIndex()

    def addPoint(self, point):
        self.suitPoints.append(point)

    def getPoint(self, pointIndex):
        return self.suitPoints[pointIndex]

    def reversePath(self):
        self.suitPoints.reverse()
        
    def __repr__(self):
        s = "DNASuitPath:\n"
        for index, point in enumerate(self.suitPoints):
            if (index + 1) < len(self.suitPoints):
                nextPoint = self.getPoint(index + 1)
            
            else:
                nextPoint = "**END**"
                
            s += "%s -> %s\n" % (point, nextPoint)
        
        return s
        
    def __len__(self):
        return self.getNumPoints()
        

File: src/suit/DNASuitPath.cxx
#include "DNASuitPath.h"

#include <algorithm>

TypeHandle DNASuitPath::_type_handle;

DNASuitPath::DNASuitPath()
{
}

DNASuitPath::~DNASuitPath()
{
}

void DNASuitPath::add_point(PT(DNASuitPoint) point)
{
    m_suit_points.push_back(point);
}

PT(DNASuitPoint) DNASuitPath::get_point(int index)
{
    return m_suit_points.at(index);
}

int DNASuitPath::get_num_points()
{
    return m_suit_points.size();
}

point_index_t DNASuitPath::get_point_index(int index)
{
    return get_point(index)->get_index();
}

void DNASuitPath::reverse_path()
{
    std::reverse(m_suit_points.begin(), m_suit_points.end());
}
